src/skfolio/measures/_measures.py

```diff
@@ -605,3 +605,23 @@ def gini_mean_difference(returns: np.ndarray) -> float:
     """
     w = owa_gmd_weights(len(returns))
     return float(w @ np.sort(returns, axis=0))
+
+
+def effective_number_assets(weights: np.ndarray) -> float:
+    """
+    Computes the effective number of assets, aka the inverse of Herfindahl index.
+
+    Parameters
+    ----------
+    weights: ndarray of shape (n_assets,)
+
+    Returns
+    -------
+    value: float
+        Effective number of assets.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Herfindahl%E2%80%93Hirschman_index#Effective_assets_in_a_portfolio
+    """
```

Comment:
We could give more details in the docstring. Also we need a space in " weights : " otherwise the docs are not formatted properly, lastly I would maybe put the original reference instead of wikipedia:

```suggestion
    r"""
    Computes the effective number of assets, defined as the inverse of the Herfindahl index [1]_:
    .. math:: N_{eff} = \frac{1}{\Vert w \Vert_{2}^{2}}
    
    It quantifies portfolio concentration, with a higher value indicating a more diversified portfolio.

    Parameters
    ----------
    weights : ndarray of shape (n_assets,)
        Weights of the assets.

    Returns
    -------
    value : float
        Effective number of assets.

    References
    ----------
    .. [1] "Banking and Financial Institutions Law in a Nutshell".
        Lovett, William Anthony (1988)
    """
```